src/Game/DrawCardButton.js

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useMemo, useState } from "react";
+import { supabase } from "../supabase_client";
+
+const DrawCardButton = (props) => {
+  const {
+    currentPlayerLocationID,
+    myPlayerData,
+    currentTurnPlayer,
+    playerName,
+    hasPlayerMovedLocations,
+    setHasPlayerDrawnCard,
+    hasPlayerDrawnCard,
+  } = props;
+  const [availableCardsList, setAvailableCardsList] = useState([]);
+
+  const fetchLocationSpecificCards = async () => {
+    const { data } = await supabase
+      .from("CardList")
+      .select("cardID, name, description")
+      .eq("locationID", currentPlayerLocationID);
+    setAvailableCardsList(data);
+  };
+
+  useEffect(() => {
+    fetchLocationSpecificCards();
+  }, [currentPlayerLocationID]);
+
+  const randomCard = useMemo(() => {
+    return availableCardsList[
+      Math.floor(Math.random() * availableCardsList.length)
+    ];
+  }, [availableCardsList]);
+
+  const sendCardToInventory = async () => {
+    await supabase
+      .from("PlayersToCardList")
+      .insert([{ playerID: myPlayerData.playerID, cardID: randomCard.cardID }]);
+    setHasPlayerDrawnCard(true);
+  };
+
+  const isMyPlayersTurn =
+    currentTurnPlayer && currentTurnPlayer.name === playerName;
```

Comment:
this is totally fine, but you should be aware of [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining). you can probably google for better examples than the mdn doc